llvm/lib/Target/AIE/AIECombinerHelper.cpp

```diff
@@ -119,6 +119,21 @@ bool llvm::canDelayMemOp(MachineInstr &MemI, MachineInstr &Dest,
   return none_of(InstrRange, UnsafeToMovePast);
 }
 
+/// \return true if \a Dest can be moved just after \a MemI in order to allow
+/// combining
+bool llvm::canAntecipateOp(MachineInstr &MemI, MachineInstr &Dest,
+                           const MachineRegisterInfo &MRI) {
```

Comment:
Maybe assert that we don't try to move an instruction (`Dest`) that has side effects or loads/stores?